cerberus/client.py

```diff
@@ -327,30 +372,69 @@ def get_secrets(self, vault_path):
             "get_secrets is deprecated, use get_secrets_data instead",
             DeprecationWarning
         )
-        secret_resp = requests.get(self.cerberus_url + '/v1/secret/' + vault_path,
-                                   headers=self.HEADERS)
+        return self._get_secrets(vault_path, version)
 
-        self.throw_if_bad_response(secret_resp)
-
-        return secret_resp.json()
-
-    def get_secrets_data(self, vault_path):
+    def get_secrets_data(self, vault_path, version=None):
         """Return json secrets based on the vault_path
 
         Keyword arguments:
 
             vault_path (string) -- full path in the secret deposit box that contains the key
         """
-        secret_resp = requests.get(self.cerberus_url + '/v1/secret/' + vault_path,
-                                   headers=self.HEADERS)
+        return self._get_secrets(vault_path, version)['data']
 
-        self.throw_if_bad_response(secret_resp)
+    def get_secret_versions(self, vault_path, limit=None, offset=None):
+        """
+        Get verions of a particular secret key
+
+        vault_path -- full path to the key in the safety deposit box
+        limit -- Default(100), limits how many records to be returned from the api at once.
+        offset -- Default(0), used for pagination.  Will request records from the given offset.
+        """
+        # Set the normal defaults
+        if(limit is None):
+            limit = 100
 
-        return secret_resp.json()['data']
+        if(offset is None):
+            offset = 0
 
+        secret_resp = requests.get(str.join('', [self.cerberus_url, '/v1/secret-versions/', vault_path,
+                                   '?limit=', str(limit), '&offset=', str(offset)]), headers=self.HEADERS)
+        self.throw_if_bad_response(secret_resp)
+        return secret_resp.json()
+
+    def get_all_secret_version_ids(self, vault_path, limit=None):
```

Comment:
For both get_all_secret_version_ids, and get_all_secret_version_ids.

I'm going back and forth on whether I should include these functions, and even if they should be kept, if they are doing things the right way.

On one hand, a generator is a good way to handle something with the possibility of returning a ton of data.  But all of the other functions in the library don't use generators, so it seems like a divergence from the rest of the style of the library.

I'm also considering whether these functions should be made "private",  until others decide if they are truly useful or not.

In the same vein, I'm wondering if something like a contrib library would be a better place to put some convenience functions until a decision is made that they should be part of the core library.

Reply:
Agreed, private is probably a good idea so that consumers don't depend on this functionality. I like the idea of generators, but I'm generally a fan of lazyness as a concept so 🤷‍♂️